src/stan/services/init/initialize_state.hpp

```diff
@@ -219,20 +212,23 @@ namespace stan {
              ++num_init_tries) {
           for (int i = 0; i < cont_params.size(); ++i)
             cont_params(i) = init_rng();
-          if (initialize_state_values(cont_params, model, output))
+          if (initialize_state_values(cont_params, model, writer))
             break;
         }
 
         if (num_init_tries > MAX_INIT_TRIES) {
-          if (output)
-            *output << std::endl << std::endl
-            << "Initialization between (" << -R << ", " << R
-            << ") failed after "
-            << MAX_INIT_TRIES << " attempts. " << std::endl
-            << " Try specifying initial values,"
-            << " reducing ranges of constrained values,"
-            << " or reparameterizing the model."
-            << std::endl;
+          std::stringstream R_ss, MAX_INIT_TRIES_ss;
```

Comment:
I don't know a better way to do it.  

If we're finding ourselves doing this a lot, we can write
our own functions to do the conversions that lean on the
streams.  The streams have the property of always using
whatever the stream's precision is for the output.

By the way, is that something we're going to control in
the writer?  The digits thing is a big deal.  I think 
C++ defaults are higher than R defaults for print.  Always
a problem with ASCII formats.  The higher precision means
more storage requirements and slower I/O.

- Bob




> On Nov 7, 2015, at 10:47 AM, bgoodri <notifications@github.com> wrote:
> 
> In src/stan/services/init/initialize_state.hpp:
> 
> >              break;
> >          }
> >  
> >          if (num_init_tries > MAX_INIT_TRIES) {
> > -          if (output)
> > -            *output << std::endl << std::endl
> > -            << "Initialization between (" << -R << ", " << R
> > -            << ") failed after "
> > -            << MAX_INIT_TRIES << " attempts. " << std::endl
> > -            << " Try specifying initial values,"
> > -            << " reducing ranges of constrained values,"
> > -            << " or reparameterizing the model."
> > -            << std::endl;
> > +          std::stringstream R_ss, MAX_INIT_TRIES_ss;
> 
> Not without C++11 AFAIK
> …
> On Sat, Nov 7, 2015 at 10:33 AM, Daniel Lee <notifications@github.com> wrote: In src/stan/services/init/initialize_state.hpp <#1671 (comment)>: > break; > } > > if (num_init_tries > MAX_INIT_TRIES) { > - if (output) > - *output << std::endl << std::endl > - << "Initialization between (" << -R << ", " << R > - << ") failed after " > - << MAX_INIT_TRIES << " attempts. " << std::endl > - << " Try specifying initial values," > - << " reducing ranges of constrained values," > - << " or reparameterizing the model." > - << std::endl; > + std::stringstream R_ss, MAX_INIT_TRIES_ss; @bob-carpenter <https://github.com/bob-carpenter> or @bgoodri <https://github.com/bgoodri>, is there a better way to output double and int as strings? — Reply to this email directly or view it on GitHub <https://github.com/stan-dev/stan/pull/1671/files#r44213437>.
> —
> Reply to this email directly or view it on GitHub.
>